addons/account/models/account_reconcile_model.py

```diff
@@ -863,6 +863,11 @@ def _get_invoice_matching_rule_result(self, st_line, candidates, aml_ids_to_excl
             # A write-off must be applied if there are some 'new' lines to propose.
             write_off_lines_vals = list(filter(lambda x: 'id' not in x, lines_vals_list))
             if not lines_vals_list or write_off_lines_vals:
+                for line_vals in write_off_lines_vals:
+                    if line_vals.get('currency_id') and line_vals.get('currency_id') != st_line.company_id.currency_id.id:
+                        currency = self.env['res.currency'].browse(line_vals['currency_id'])
+                        for key in ('debit', 'credit', 'balance'):
+                            line_vals[key] = st_line.company_id.currency_id._convert(line_vals[key], currency, st_line.company_id, st_line.date, False)
```

Comment:
I'm just having a second thought: doesn't it modify what amls are created later on?
Can we check that functionally together when you have time?

(and you seem to tell to LAS that all the computation already happened at that moment. I want to discuss that a bit with you)

Reply:
"doesn't it modify what amls are created later on?"
- As it's the last method called before sending the values to the widget, I think not.

"you seem to tell to LAS that all the computation already happened at that moment"
- What I tried to explain is that even with setting 'currency_id' here https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_reconcile_model.py#L368
the issue was still remain, as we sent the wrong amount in the method, and no conversion was made after that

Reply:
@guva-odoo I faced the same issue on my refactoring. You should use `_prepare_counterpart_move_line_vals` and add a comment. See my "fix"
https://github.com/odoo/enterprise/compare/master...odoo-dev:master-bank-st-poc-4-las?expand=1#diff-a46ffb6bf25aac6eb588af8b4786716b7a2d0e23cc9cb7c5b8ce8b84ac886c03R1055